crates/compression/src/ports.rs

```diff
@@ -18,20 +18,46 @@ use fuel_core_types::{
 /// must be committed atomically, after which block height must be incremented.
 pub trait TemporalRegistry<T> {
     /// Reads a value from the registry at its current height.
-    fn read_registry(&self, key: RegistryKey) -> anyhow::Result<T>;
+    fn read_registry(&self, key: &RegistryKey) -> anyhow::Result<T>;
 
     /// Reads a value from the registry at its current height.
-    fn write_registry(&mut self, key: RegistryKey, value: &T) -> anyhow::Result<()>;
+    fn write_registry(&mut self, key: &RegistryKey, value: &T) -> anyhow::Result<()>;
```

Comment:
I don't understand why `RegistryKey` should be passed by reference. It's small enough to fit into a register so performance is likely not the reason here.

Reply:
Just to avoid `*key` and `&key` in the code=) You can see that now everything just `key`